src/definition.cpp

```diff
@@ -545,7 +545,7 @@ void DefinitionImpl::writeDocAnchorsToTagFile(FTextStream &tagFile) const
       {
         //printf("write an entry!\n");
         if (definitionType()==TypeMember) tagFile << "  ";
-        tagFile << "    <docanchor file=\"" << si->fileName << "\"";
+        tagFile << "    <docanchor file=\"" << si->fileName << (hasExtension(si->fileName) ? "" : Doxygen::htmlFileExtension) << "\"";
```

Comment:
Wouldn't it be cleaner to instead of adding `hasExtension()`, add a `addHtmlExtension(fileName)` function that returns a adjusted file name?

Reply:
The name `addHtmlExtension` is not completely covering the purpose, the `HTML_FILE_EXTENSION` will be added only in case no extension is present. This is why I chose for the given construct. I think a better routine name for `addHtmlExtension` would be beneficial.

Notwithstanding the above:
You wrote "add a addHtmlExtension(fileName) function that returns a adjusted file name"

Would that be a function like
```
QCString addHtmlExtension(QCString fName)
{
  return ((QFileInfo(fName).extension(FALSE).isEmpty()) ? Doxygen::htmlFileExtension : "");
}
```
and in the code 
```
t << f << addHtmlExtension(f)
```

or
```
QCString addHtmlExtension(QCString fName)
{
  return fName + ((QFileInfo(fName).extension(FALSE).isEmpty()) ? Doxygen::htmlFileExtension : "");
}
```
with in the code
```
t << addHtmlExtension(f)
```